src/main/java/com/google/sps/servlets/NaturalLanguageServlet.java

```diff
@@ -63,9 +100,33 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
         if (this.nlp == null) {
             this.nlp = new NaturalLanguageProcessor();
         }
+        if (this.db == null) {
+            this.db = new CaptionStorage();
+        }
 
         // Builds the Java object from JSON and preprocesses the captions by redefining time ranges
         YoutubeCaptions youtubeCaptions = gson.fromJson(json, YoutubeCaptions.class);
+
+        // Parses the video ID from the url
+        boolean addToDatabase = true;
+        String videoID = "";
+        String[] urlParts = youtubeCaptions.getVideoURL().split(VIDEO_URL_ID_DELIMITER);
+        if (urlParts.length <= 1) {
+            addToDatabase = false;
+        }
+        if (addToDatabase) {
+            videoID = urlParts[1];
```

Comment:
it is possible that urlParts.legnth is only 1 here right?

Reply:
In the front end, we check that the url matches a regex before passing it into the backend, so in that use case, `urlParts.length` should be greater than 1. However, it is possible that `urlParts.length` is 1 when running mock captions or when testing. In this case, the `addToDatabase` boolean will be false, so nothing is added to the database and we won't access anything out of bounds of `urlParts`.